tensorflow_quantum/datasets/spin_system_test.py

```diff
@@ -93,23 +82,126 @@ def test_returned_objects(self):
+                self.assertAllClose(e, addinfo[n].gs_energy, rtol=5e-3)
+
+    def test_returned_objects(self):
         """Test that the length and types of returned objects are correct."""
-        supported_nspins = [4, 8, 12, 16]
-        for nspins in supported_nspins:
-            qbs = cirq.GridQubit.rect(nspins, 1)
-            circuits, _, _, addinfo = spin_system.tfi_chain(qbs, 'closed')
-            for n in range(81):
+        for nspins in SUPPORTED_NSPINS_XXZ_CHAIN:
+            circuits, labels, pauli_sums, addinfo = DATA_DICT_XXZ_CHAIN[nspins]
+            self.assertLen(circuits, 76)
+            self.assertLen(labels, 76)
+            self.assertLen(pauli_sums, 76)
+            self.assertLen(addinfo, 76)
+            for n in range(76):
+                self.assertIsInstance(circuits[n], cirq.Circuit)
+                self.assertIsInstance(labels[n], int)
+                self.assertIsInstance(pauli_sums[n], cirq.PauliSum)
+                self.assertIsInstance(addinfo[n], SpinSystemInfo)
+
+    def test_param_resolver(self):
+        """Test that the resolved circuits are correct."""
+        for nspins in SUPPORTED_NSPINS_XXZ_CHAIN:
+            circuits, _, _, addinfo = DATA_DICT_XXZ_CHAIN[nspins]
+            for n in RANDOM_SUBSET_XXZ_CHAIN:
                 resolved_circuit = cirq.resolve_parameters(
                     addinfo[n].var_circuit, addinfo[n].params)
                 state_circuit = cirq.Simulator().simulate(
                     circuits[n]).final_state
                 state_resolved_circuit = cirq.Simulator().simulate(
                     resolved_circuit).final_state
                 self.assertAllClose(np.abs(
-                    np.conj(np.dot(state_circuit, state_resolved_circuit))),
+                    np.vdot(state_circuit, state_resolved_circuit)),
                                     1.0,
                                     rtol=1e-3)
 
 
 if __name__ == '__main__':
+    RANDOM_SUBSET_SIZE = 10
+    # TFI CHAIN
+    SUPPORTED_NSPINS_TFI_CHAIN = [4, 8, 12, 16]
+    DATA_DICT_TFI_CHAIN = {}
+    QBS_DICT_TFI_CHAIN = {}
+    for nspins in SUPPORTED_NSPINS_TFI_CHAIN:
+        QBS_TFI_CHAIN = cirq.GridQubit.rect(nspins, 1)
+        DATA_DICT_TFI_CHAIN[nspins] = spin_system.tfi_chain(
+            QBS_TFI_CHAIN,
+            'closed',
+        )
+        QBS_DICT_TFI_CHAIN[nspins] = QBS_TFI_CHAIN
+    RANDOM_SUBSET_TFI_CHAIN = np.random.permutation(list(
+        range(76)))[:RANDOM_SUBSET_SIZE]
+    # XXZ CHAIN
+    SUPPORTED_NSPINS_XXZ_CHAIN = [4, 8, 12, 16]
+    DATA_DICT_XXZ_CHAIN = {}
+    QBS_DICT_XXZ_CHAIN = {}
+    for nspins in SUPPORTED_NSPINS_XXZ_CHAIN:
+        QBS_XXZ_CHAIN = cirq.GridQubit.rect(nspins, 1)
+        DATA_DICT_XXZ_CHAIN[nspins] = spin_system.xxz_chain(
+            QBS_XXZ_CHAIN,
+            'closed',
+        )
+        QBS_DICT_XXZ_CHAIN[nspins] = QBS_XXZ_CHAIN
+    RANDOM_SUBSET_XXZ_CHAIN = np.random.permutation(list(
+        range(76)))[:RANDOM_SUBSET_SIZE]
```

Comment:
Can we remove this so we just call `tf.test.main()` ?

Reply:
I moved this to below the imports. I can't figure out another way to make sure that the data loader is called only once when testing.

Reply:
Would something like this work ? 

https://www.tensorflow.org/api_docs/python/tf/test/TestCase#setUpClass

Reply:
Yes, but then pylint complains that the method names are not correct.

Reply:
I added an exception for pylint to ignore method names for both tests.